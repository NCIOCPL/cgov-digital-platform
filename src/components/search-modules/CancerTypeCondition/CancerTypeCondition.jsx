import React from 'react';
import { Fieldset, TextInput } from '../../atomic';
import './CancerTypeCondition.scss';

const CancerTypeCondition = ({ handleUpdate, useValue }) => {
  return (
    <Fieldset
      id="type"
      legend="Cancer Type/Condition"
      helpUrl="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#cancertype"
    >
<<<<<<< HEAD
      <TextInput
        action={handleUpdate}
        id="cancerTypeCondition"
        name="cancerTypeCondition"
        value={useValue('cancerTypeCondition')}
        type="text"
      />
=======
      <p>
        Select a cancer type or condition. Select additional options, if
        applicable.
      </p>
      <TextInput id="q" label="Primary Cancer Type/Condition" />
      <TextInput id="st" label="Subtype" />
      <TextInput id="stage" label="Stage" />
      <TextInput id="fin" label="Side Effects/Biomarkers/Participant Attributes" />
>>>>>>> 0d6359b5864272369127a33c3a4609d19f3ca437
    </Fieldset>
  );
};

export default CancerTypeCondition;
