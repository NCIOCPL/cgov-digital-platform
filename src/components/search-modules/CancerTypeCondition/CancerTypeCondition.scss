.cancer-type-condition {
  .subsearch {
    margin-top: 1rem;

    &.disabled {
      position: relative;
      &::after {
        content: '';
        display: block;
        position: absolute;
        background: white;
        top: 0;
        left: 0;
        height: 100%;
        width: 100%;
        opacity: 0.5;
      }
    }
  }

  .ct-select {
    position: relative;

    &__button {
      appearance: none;
      background-color: white;
      box-shadow: none;
      border: 1px solid #bdbdc2;
      color: #2e2e2e;
      font-size: 1em;
      font-family: Noto Sans, Century Gothic, Arial, sans-serif;
      font-weight: 400;
      padding: 0.5em;
      margin-bottom: 1em;
      display: block;
      width: 100%;
      text-align: left;
      background-image: url('data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAxMCAxNCI+PGcgZmlsbD0iIzNkNDU1MSIgZmlsbC1ydWxlPSJldmVub2RkIj48cGF0aCBkPSJNLjYyNjM5NjI4IDUuOTk5OTM4OTZIOS4zNzI3Mzg5Yy4yNTMzOTE3My4wMDEzMDg1Ni40ODI0MTYtLjE1MDc0MjY0LjU3OTU2MjAyLS4zODQ3NzYyNC4wOTcxNDU5OC0uMjM0MDMzNi4wNDMxMjI5NC0uNTAzNTc2MTYtLjEzNjcwOTIyLS42ODIwOTY0TDUuNDQwMTgxNS4xODM1NDM1QzUuMzI0NzU2MDYuMDY2MTMxMTYgNS4xNjcwMDgzIDAgNS4wMDIzNjExNCAwYy0uMTY0NjQ3MTYgMC0uMzIyMzk0OS4wNjYxMzExNy0uNDM3ODIwMzcuMTgzNTQzNUwuMTgzNTQzNSA0LjkzMzA2NjMyQy4wNjYxMzExNiA1LjA0ODQ5MTggMCA1LjIwNjIzOTUzIDAgNS4zNzA4ODY3cy4wNjYxMzExNy4zMjIzOTQ5Mi4xODM1NDM1LjQzNzgyMDM4Yy4xMTQ2NDQ0LjEyMjQ5MzQuMjc1MDgwMDYuMTkxNzcyNDQuNDQyODUyNzguMTkxMjMxODh6TTkuMzcyNzM4OSA3Ljk5OTk2ODVILjYyNjM5NjNjLS4xNjYyNjUzLS4wMDMxNjE4My0uMzI2NDc2MDguMDYyMzc4OTQtLjQ0Mjg1MjguMTgxMTY3MDVDLjA2NjEzMTIgOC4yOTY1NjEgMCA4LjQ1NDMwODc1IDAgOC42MTg5NTU5MnMuMDY2MTMxMTcuMzIyMzk0OTIuMTgzNTQzNS40Mzc4MjAzN2w0LjM4MDk5NzI2IDQuNzU5NjgwMkM0LjY3OTk2NjIyIDEzLjkzMzg2ODggNC44Mzc3MTM5NiAxNCA1LjAwMjM2MTEzIDE0cy4zMjIzOTQ5Mi0uMDY2MTMxMi40Mzc4MjAzOC0uMTgzNTQzNWw0LjM3NTQxMDItNC43NDk2MTUzN2MuMTc5ODMyMTYtLjE3ODUyMDIzLjIzMzg1NTItLjQ0ODA2Mjc4LjEzNjcwOTIyLS42ODIwOTYzOC0uMDk3MTQ2MDItLjIzNDAzMzYtLjMyNjE3MDMtLjM4NjA4NDgtLjU3OTU2Mi0uMzg0Nzc2MjV6Ii8+PC9nPjwvc3ZnPg=='),
        linear-gradient(transparent, transparent);
      background-repeat: no-repeat;
      background-position: right 0.75rem center;
      background-size: 0.5rem;

      &.--modified {
        background-color: rgba(254, 230, 133, .4);
      }
    }
    &__menu {
      display: none;
      position: absolute;
      width: 100%;
      background: white;
      top: 68px;
      left: 0;
      z-index: 10;
      padding: 1rem;
      box-shadow: 0 0 4px rgba(0, 0, 0, 0.2);

      &.open {
        display: block;
      }

      .cts-autocomplete__menu {
        position: relative;
        top: 1px;
      }
    }
  }
}
