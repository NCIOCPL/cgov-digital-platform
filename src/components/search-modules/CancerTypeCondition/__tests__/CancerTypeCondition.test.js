import React from 'react';
import { render, shallow, mount } from 'enzyme';
import {
  resetDom,
  setupDom
} from '../../../../utilities/testHelpers';
import CancerTypeCondition from '../CancerTypeCondition';

const defaultProps = {
}

const setup = (enzymeMethod = render, props = {}) => {
  const component = enzymeMethod(
    <CancerTypeCondition {...props} />
  );
  return component;
};

describe('CancerTypeCondition', () => {
  describe('Render', () => {
    let component = null;

    afterEach(() => {
      resetDom();
    });

    it('renders without error', () => {
      component = setup(render, defaultProps);
      expect(component).toMatchSnapshot();
    });
  });

});
