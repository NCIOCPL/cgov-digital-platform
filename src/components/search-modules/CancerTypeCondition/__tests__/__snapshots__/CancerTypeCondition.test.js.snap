// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CancerTypeCondition Render renders without error 1`] = `
<fieldset
  class="cts-fieldset "
  id="fieldset--type"
>
  <legend
    class="cts-fieldset__legend"
  >
    <span>
      Cancer Type/Condition
    </span>
  </legend>
  <a
    aria-label="Help"
    class="text-icon-help"
    href="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#cancertype"
    rel="noopener noreferrer"
    target="_blank"
  >
    ?
  </a>
  <div
    class="cts-fieldset__body"
  >
<<<<<<< HEAD
=======
    <p>
      Select a cancer type or condition. Select additional options, if applicable.
    </p>
>>>>>>> 0d6359b5864272369127a33c3a4609d19f3ca437
    <div
      class=""
    >
      <label
        class="cts-label"
<<<<<<< HEAD
        for="cancerTypeCondition"
        id="cancerTypeCondition-label"
      />
      <input
        aria-labelledby="cancerTypeCondition-label"
        aria-required="false"
        class="cts-input null"
        id="cancerTypeCondition"
        name="cancerTypeCondition"
=======
        for="q"
        id="q-label"
      >
        Primary Cancer Type/Condition
      </label>
      <input
        aria-labelledby="q-label"
        aria-required="false"
        class="cts-input  "
        id="q"
        spellcheck="false"
        type="text"
        value=""
      />
    </div>
    <div
      class=""
    >
      <label
        class="cts-label"
        for="st"
        id="st-label"
      >
        Subtype
      </label>
      <input
        aria-labelledby="st-label"
        aria-required="false"
        class="cts-input  "
        id="st"
        spellcheck="false"
        type="text"
        value=""
      />
    </div>
    <div
      class=""
    >
      <label
        class="cts-label"
        for="stage"
        id="stage-label"
      >
        Stage
      </label>
      <input
        aria-labelledby="stage-label"
        aria-required="false"
        class="cts-input  "
        id="stage"
        spellcheck="false"
        type="text"
        value=""
      />
    </div>
    <div
      class=""
    >
      <label
        class="cts-label"
        for="fin"
        id="fin-label"
      >
        Side Effects/Biomarkers/Participant Attributes
      </label>
      <input
        aria-labelledby="fin-label"
        aria-required="false"
        class="cts-input  "
        id="fin"
>>>>>>> 0d6359b5864272369127a33c3a4609d19f3ca437
        spellcheck="false"
        type="text"
        value=""
      />
    </div>
  </div>
</fieldset>
`;
